spec/models/manageiq/providers/hawkular/middleware_manager/event_catcher_spec.rb

```diff
@@ -0,0 +1,50 @@
+describe ManageIQ::Providers::Hawkular::MiddlewareManager::EventCatcher do
+  before do
+    allow(MiqServer).to receive(:my_zone).and_return("default")
+    auth = AuthToken.new(:name => "test", :auth_key => "valid-token", :userid => "jdoe", :password => "password")
+    @ems = FactoryGirl.create(:ems_hawkular,
+                              :hostname        => 'localhost',
+                              :port            => 8080,
+                              :authentications => [auth])
+
+    WebMock.allow_net_connect!
+  end
+
+  include ManageIQ::Providers::Hawkular::MiddlewareManager::EventCatcherMixin
+
+  context "#perf_catch_events" do
+    # VCR.eject_cassette
+    # VCR.turn_off!(ignore_cassettes: true)
+
+    VCR.configure do |c|
+      c.default_cassette_options = {
+        :match_requests_on => [:method, VCR.request_matchers.uri_without_params(:startTime, :endTime)]
+      }
+    end
+
+    it "will invoke a perf_catch_events" do
+      VCR.use_cassette(described_class.name.underscore.to_s) do # , :record => :new_episodes) do
+        def perf_catch_events(start_time)
+          # puts "Collecting Events for interval [Start=#{start_time}]"
+
+          Benchmark.realtime_block(:collect_data) do
+            begin
+              result = []
+              event_monitor_handle(start_time).each do |event|
+                # puts "Handling Event #{event}"
+                result.push(event)
+              end
+              expect(result.count).to be == 1
+              expect(result[0].category).to eq 'Hawkular Deployment'
+            rescue => e
+              puts "Hawkular alerts service unavailable: #{e.message}"
+              return [{}, {}]
+            end
+          end
+        end
+
+        perf_catch_events(90.minutes.ago.beginning_of_minute.to_i * 1000)
```

Comment:
right, up above in the  `VCR.configure` block I set it up such that we ignore the startTime and endTime when matching the cassette.  But if we run without a cassette or generate a new one then we'd need it.  So, you're right, it's not really required as is, but because the start time here (90 minutes back) is different than the default, I pass it in.